src/AutoRest.CSharp/Common/Input/Source/SourceInputModel.cs

```diff
@@ -32,14 +35,19 @@ public SourceInputModel(Compilation compilation)
 
             foreach (IModuleSymbol module in assembly.Modules)
             {
-                foreach (var type in GetSymbols(module.GlobalNamespace))
+                foreach (var type in SourceInputHelper.GetSymbols(module.GlobalNamespace))
                 {
                     if (type is INamedTypeSymbol namedTypeSymbol && TryGetName(type, out var schemaName))
                     {
                         _nameMap.Add(schemaName, namedTypeSymbol);
                     }
                 }
             }
+
+            if (existingCompilation != null)
+            {
+                existingCompilation.FilterSymbols();
+            }
```

Comment:
I believe this method on this instance should only be invoke once, therefore it never should be public and we should not invoke it here. It should be invoked internally in this instance (for instance when first time we call the FindSymbol on it)